packages/image_picker/image_picker_ios/test/image_picker_ios_test.dart

```diff
@@ -0,0 +1,937 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:image_picker_ios/image_picker_ios.dart';
+import 'package:image_picker_ios/src/messages.g.dart';
+import 'package:image_picker_platform_interface/image_picker_platform_interface.dart';
+
+import 'test_api.dart';
+
+@immutable
+class _LoggedMethodCall {
+  const _LoggedMethodCall(this.name, {required this.arguments});
+  final String name;
+  final Map<String, Object?> arguments;
+
+  @override
+  bool operator ==(Object other) {
+    return other is _LoggedMethodCall &&
+        name == other.name &&
+        mapEquals(arguments, other.arguments);
+  }
+
+  @override
+  int get hashCode => Object.hash(name, arguments);
+
+  @override
+  String toString() {
+    return 'MethodCall: $name $arguments';
+  }
+}
+
+class _ApiLogger implements TestHostImagePickerApi {
```

Comment:
Can't you use a mocking library for this?

Reply:
Under this testing structure, a mocking library doesn't help; I'm already implementing all of the methods to log them anyway.

If I completely restructured the tests, then yes, but as mentioned in my initial PR comments I deliberately changed the tests as little as I could in this PR in order to ensure that implementations didn't regress something in a way that was masked by test change mistakes.